### PR TITLE
Migrate to GOV.UK Frontend v3

### PR DESCRIPTION
**Pull request by freesteph**
# Description
Migrating involved updating a lot of markup to conform to the new classes, and anything pointed out in the [migration guide](https://github.com/alphagov/govuk-frontend/releases/tag/v3.0.0). The reader is advised to read the commit log for finer details and more importantly building the branch for some exploratory testing.

## Steps to run: 
1. `git fetch origin`;
2. `git checkout origin/migrate-to-govuk-frontend`;
3. `make serve`
4. Visit `http://localhost:4567`.

### CODE REVIEW

File: source/layouts/layout.erb [hunk 1]
```diff
@@ -111,27 +126,40 @@
...
+          <div class="govuk-footer__meta-item">
+            <a class="govuk-footer__link govuk-footer__copyright-logo" href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/">© Crown copyright</a>
           </div>
         </div>
       </div>
     </footer>
+    <script type="text/javascript">
+      document.body.className
+        ? document.body.className += " js-enabled"
+        : document.body.className = " js-enabled";
```
Comment (robinmitra) [inline source/layouts/layout.erb]:
```suggestion
        : document.body.className = "js-enabled";
```

File: source/index.html.erb [hunk 2]
```diff
@@ -2,50 +2,67 @@
...
-      <p>Sign up to connect your device to the GovWifi network across the public sector.</p>
-      <%= link_to "Connect a device", "/about-govwifi/connect-to-govwifi/", class: "govuk-link bold" %>
-    </div>
+  <div class="govuk-width-container">
+    <div class="govuk-main-wrapper govuk-main-wrapper--l">
+      <div class="govuk-grid-row">
+        <div class="govuk-grid-column-one-third">
+          <h2 class="govuk-heading-l">Connect a device</h2>
+          <p class="govuk-body">Sign up to connect your device to the GovWifi network across the public sector.</p>
+          <p class="govuk-body govuk-!-margin-bottom-8">
```
Comment (robinmitra) [inline source/index.html.erb]: `govuk-!-margin-bottom-8` should ideally be added to the row element (i.e. line 38)

File: source/index.html.erb [hunk 3]
```diff
@@ -2,50 +2,67 @@
...
-    <div class="column-one-third">
-      <h2>Offer GovWifi</h2>
-      <p>Make GovWifi available in your organisation and keep your existing infrastructure and wifi provider.</p>
-      <%= link_to "Offer GovWifi", "https://docs.wifi.service.gov.uk", class: "govuk-link bold" %>
-    </div>
+        <div class="govuk-grid-column-one-third">
+          <h2 class="govuk-heading-l">Offer GovWifi</h2>
+          <p class="govuk-body">Make GovWifi available in your organisation and keep your existing infrastructure and
+            wifi provider.</p>
+          <p class="govuk-body govuk-!-margin-bottom-8">
```
Comment (robinmitra) [inline source/index.html.erb]: Same as above.

File: source/index.html.erb [hunk 4]
```diff
@@ -2,50 +2,67 @@
...
+        </div>
 
-    <div class="column-one-third">
-      <h2>Manage GovWifi</h2>
-      <p>Manage an existing GovWifi installation in your organisation as an administrator.</p>
-      <%= link_to "Network administrator login", "https://admin.wifi.service.gov.uk", class: "govuk-link bold" %>
+        <div class="govuk-grid-column-one-third">
+          <h2 class="govuk-heading-l">Manage GovWifi</h2>
+          <p class="govuk-body">Manage an existing GovWifi installation in your organisation as an administrator.</p>
+          <p class="govuk-body govuk-!-margin-bottom-8">
```
Comment (robinmitra) [inline source/index.html.erb]: Same as above.

File: source/stylesheets/_core.scss [hunk 5]
```diff
@@ -1,36 +1,39 @@
...
+//@import "govuk-elements-sass/public/sass/elements/elements-typography";
+//@import "govuk-elements-sass/public/sass/elements/forms";
+//@import "govuk-elements-sass/public/sass/elements/layout";
+//@import "govuk-elements-sass/public/sass/elements/lists";
+//@import "govuk-elements-sass/public/sass/elements/panels";
+//@import "govuk-elements-sass/public/sass/elements/tables";
+
+//@import "govuk_template/source/assets/stylesheets/styleguide/colours";
+//@import "govuk_template/source/assets/stylesheets/accessibility";
+//@import "govuk_template/source/assets/stylesheets/footer";
```
Comment (phantas) [inline source/stylesheets/_core.scss]: remove comments

File: source/stylesheets/manifest.css.scss [hunk 6]
```diff
@@ -1,18 +1 @@
...
-
-//= link govuk_template/source/assets/stylesheets/images/close.png
-//= link govuk_template/source/assets/stylesheets/images/gov.uk_logotype_crown-1x.png
-//= link govuk_template/source/assets/stylesheets/images/gov.uk_logotype_crown.png
-//= link govuk_template/source/assets/stylesheets/images/govuk-crest-2x.png
-//= link govuk_template/source/assets/stylesheets/images/govuk-crest-ie.png
-//= link govuk_template/source/assets/stylesheets/images/govuk-crest.png
-//= link govuk_template/source/assets/stylesheets/images/open-government-licence.png
-//= link govuk_template/source/assets/stylesheets/images/open-government-licence_2x.png
+//= link_tree ../../node_modules/govuk-frontend/govuk/assets
```
Comment (phantas) [inline source/stylesheets/manifest.css.scss]: Can't we delete this file? Looks empty

File: source/stylesheets/modules/_breadcrumbs.scss [hunk 7]
```diff
@@ -26,75 +26,76 @@
...
+//      a:visited {
+//        color: $white;
+//      }
+//
+//      a:focus {
+//        color: $black;
+//      }
+//    }
+//  }
+//}
```
Comment (phantas) [inline source/stylesheets/modules/_breadcrumbs.scss]: we can delete this file too

File: source/stylesheets/modules/_content-section.scss [hunk 8]
```diff
@@ -10,19 +10,19 @@
...
+//
+//  &--with-top-border {
+//    padding-top: $gutter;
+//    border-top: 1px $grey-3 solid;
+//
+//    .content-section__first-heading {
+//      margin-top: 0;
+//    }
+//  }
+//}
```
Comment (phantas) [inline source/stylesheets/modules/_content-section.scss]: and delete this file too

File: source/stylesheets/modules/_task-list.scss [hunk 9]
```diff
@@ -53,15 +53,15 @@ $task-list-indent: 35px;
...
 
 
-.task-name {
-  width: $two-thirds;
-  float: left;
-}
+//.task-name {
+//  width: $two-thirds;
+//  float: left;
+//}
```
Comment (phantas) [inline source/stylesheets/modules/_task-list.scss]: delete comments

**Pull request merged**